substrate/runtime/src/lib.rs

```diff
@@ -603,4 +720,21 @@ sp_api::impl_runtime_apis! {
       AuthorityDiscovery::authorities()
     }
   }
+
+  impl dex::AssetConversionApi<
+    Block,
+    SubstrateAmount,
+    SubstrateAmount,
+    Coin,
+  > for Runtime {
+    fn quote_price_exact_tokens_for_tokens(asset1: Coin, asset2: Coin, amount: SubstrateAmount, include_fee: bool) -> Option<SubstrateAmount> {
+      Dex::quote_price_exact_tokens_for_tokens(asset1, asset2, amount, include_fee)
+    }
+    fn quote_price_tokens_for_exact_tokens(asset1: Coin, asset2: Coin, amount: SubstrateAmount, include_fee: bool) -> Option<SubstrateAmount> {
+      Dex::quote_price_tokens_for_exact_tokens(asset1, asset2, amount, include_fee)
+    }
+    fn get_reserves(asset1: Coin, asset2: Coin) -> Option<(SubstrateAmount, SubstrateAmount)> {
+      Dex::get_reserves(&asset1, &asset2).ok()
+    }
+  }
```

Comment:
I think this only has benefit if you additionally pull in the RPC hooks for the DEX.